WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadActivity.kt

```diff
@@ -21,52 +34,94 @@ class BackupDownloadActivity : LocaleAwareActivity() {
         setContentView(R.layout.backup_download_activity)
 
         setSupportActionBar(toolbar_main)
-        supportActionBar?.let {
-            it.setHomeButtonEnabled(true)
-            it.setDisplayHomeAsUpEnabled(true)
-            // todo: annmarie - this is here to get past lint - this will be set from childfrag
-            it.title = getString(R.string.backup_download_details_page_title)
-        }
-        setupViewModel()
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        initViewModel(savedInstanceState)
     }
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         if (item.itemId == android.R.id.home) {
             onBackPressed()
             return true
         }
-        return super.onOptionsItemSelected(item)
+        return false
+    }
+
+    override fun onBackPressed() {
+        viewModel.onBackPressed()
     }
 
-    private fun setupViewModel() {
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        viewModel.writeToBundle(outState)
+    }
+
+    private fun initViewModel(savedInstanceState: Bundle?) {
         viewModel = ViewModelProvider(this, viewModelFactory)
                 .get(BackupDownloadViewModel::class.java)
-        viewModel.start()
 
-        // todo: annmarie temporary start
-        val fragment = BackupDownloadDetailsFragment.newInstance()
-        showFragment(fragment, BackupDownloadDetailsFragment.TAG, slideIn = false, isRootFragment = true)
+        viewModel.navigationTargetObservable
+                .observe(this, { target ->
+                    target?.let {
+                        showStep(target)
+                    }
+                })
+
+        viewModel.screenTitleObservable.observe(this, { title ->
+            supportActionBar?.title = getString(title)
+            // Change the activity title for accessibility purposes
+            this.title = getString(title)
+        })
+
+        // todo: annmarie - may there are more states?
+        // Canceled, Running, Complete -> (Running = kick off status)
+        viewModel.wizardFinishedObservable.observe(this, { backupDownloadWizardState ->
+            backupDownloadWizardState?.let {
+                val intent = Intent()
+                val (backupDownloadCreated, activityId) = when (backupDownloadWizardState) {
+                    // teh request was canceled
+                    is BackupDownloadCanceled -> Pair(false, null)
+                    is BackupDownloadInProgress -> {
```

Comment:
Minor (🔍 ): Consider refactor this to a one-liner to be consistent with the rest. The comment can then be moved to the when value instead, something like:
```
// The request is in progress and user didn't want to wait around
is BackupDownloadInProgress -> Pair(true, backupDownloadWizardState.activityId)
                    
```

Reply:
in 211c480